### PR TITLE
Fix pagination 2

### PR DESCRIPTION
**Pull request by kicferk1**
## Ticket and context

Turns out adding pagination can be tricky. I will play around with the review app to make sure it behaves the way I expect it.

### Is there anything weird that code reviewer should know?
I think the real fix is removing front slashes from R&P urls. But just in case I added some more safety measures, like `sleep(1)` and checking whether we got any new R&P ids. 

### Review Checks
- [x] All commit messages are meaningful and true

### CODE REVIEW

File: app/services/register_and_partner_api/sync_users.rb [hunk 1]
```diff
@@ -20,27 +20,39 @@ def self.perform(all: false)
...
       Rails.logger.warn("Failed to sync users: #{e}")
     end
 
+    def self.perform_pagination(base_query)
+      is_last_page = false
+      page_number = 0
+      all_user_ids = []
+      until is_last_page
+        page_number += 1
+        paginated_query = base_query.merge(page: { page: page_number, per_page: 10 })
```
Comment (ethanmills) [inline app/services/register_and_partner_api/sync_users.rb:36]: Why 10 per page?

**Review by ethanmills — APPROVED** [review]
I'm wondering if there's an edge case here where we miss some users, if they're created at the wrong moment, or if there's some clock drift between the two servers. Do you think there would be any harm in `updated_since: last_sync - 5.minutes` just for a bit of margin?

**Comment by ethanmills** [comment]
Just bear in mind that at 10 per page, 1 page per second, initial sync is going to take longer than 5 mins. Will this cause issues?

**Comment by kicferk1** [comment]
> Just bear in mind that at 10 per page, 1 page per second, initial sync is going to take longer than 5 mins. Will this cause issues?

Oh, my bad, that should be 100. Good spot!

File: app/services/register_and_partner_api/sync_users.rb (hunk 1 above)
Comment (kicferk1) [reply app/services/register_and_partner_api/sync_users.rb:36]: Cause it made testing easier and I forgot to take it out.

**Comment by kicferk1** [comment]
> I'm wondering if there's an edge case here where we miss some users, if they're created at the wrong moment, or if there's some clock drift between the two servers. Do you think there would be any harm in `updated_since: last_sync - 5.minutes` just for a bit of margin?

Eh, that's what the `all` flag is for - we sync all users at night.
I think it's pretty unlikely we miss anything this way - but concurrency is a pain.

**Pull request merged**